lib/src/compile.ts

```diff
@@ -2,356 +2,51 @@
 // MIT-style license that can be found in the LICENSE file or at
 // https://opensource.org/licenses/MIT.
 
-import * as p from 'path';
-import {Observable} from 'rxjs';
-import * as supportsColor from 'supports-color';
-
-import * as proto from './vendor/embedded_sass_pb';
-import * as utils from './utils';
-import {AsyncEmbeddedCompiler} from './async-compiler';
-import {CompileResult, Options, SourceSpan, StringOptions} from './vendor/sass';
-import {Dispatcher, DispatcherHandlers} from './dispatcher';
-import {Exception} from './exception';
-import {FunctionRegistry} from './function-registry';
-import {ImporterRegistry} from './importer-registry';
-import {MessageTransformer} from './message-transformer';
-import {PacketTransformer} from './packet-transformer';
-import {SyncEmbeddedCompiler} from './sync-compiler';
-import {deprotofySourceSpan} from './deprotofy-span';
-import {
-  removeLegacyImporter,
-  removeLegacyImporterFromSpan,
-  legacyImporterProtocol,
-} from './legacy/utils';
-
-/// Allow the legacy API to pass in an option signaling to the modern API that
-/// it's being run in legacy mode.
-///
-/// This is not intended for API users to pass in, and may be broken without
-/// warning in the future.
-type OptionsWithLegacy<sync extends 'sync' | 'async'> = Options<sync> & {
-  legacy?: boolean;
-};
-
-/// Allow the legacy API to pass in an option signaling to the modern API that
-/// it's being run in legacy mode.
-///
-/// This is not intended for API users to pass in, and may be broken without
-/// warning in the future.
-type StringOptionsWithLegacy<sync extends 'sync' | 'async'> =
-  StringOptions<sync> & {legacy?: boolean};
+import {initAsyncCompiler} from './async-compiler';
+import {OptionsWithLegacy, StringOptionsWithLegacy} from './compiler';
+import {initCompiler} from './sync-compiler';
+import {CompileResult} from './vendor/sass';
 
 export function compile(
   path: string,
   options?: OptionsWithLegacy<'sync'>
 ): CompileResult {
-  const importers = new ImporterRegistry(options);
-  return compileRequestSync(
-    newCompilePathRequest(path, importers, options),
-    importers,
-    options
-  );
+  const compiler = initCompiler();
+  const result = compiler.compile(path, options);
+  compiler.dispose();
+  return result;
 }
 
 export function compileString(
   source: string,
   options?: StringOptionsWithLegacy<'sync'>
 ): CompileResult {
-  const importers = new ImporterRegistry(options);
-  return compileRequestSync(
-    newCompileStringRequest(source, importers, options),
-    importers,
-    options
-  );
+  const compiler = initCompiler();
+  const result = compiler.compileString(source, options);
+  compiler.dispose();
```

Comment:
I like that this essentially turns the top level `compile*` methods into convenience methods for accessing the compiler, meaning it always goes through the same steps.

Nothing stuck out to me, but I want to verify- does this approach add any new spinup steps to the top level `compile*` functions that are only really needed for the Compiler? I want to avoid any unnecessary performance hit here.